h2o-automl/src/test/java/ai/h2o/automl/AutoMLTest.java

```diff
@@ -138,27 +140,61 @@
     }
   }
 
-
-  @Ignore //reenable in PUBDEV-5956
-  @Test public void KeepCrossValidationFoldAssignmentTest() {
-    AutoML aml=null;
-    Frame fr=null;
+  @Test public void KeepCrossValidationFoldAssignmentEnabledTest() {
+    AutoML aml = null;
+    Frame fr = null;
+    Model leader = null;
     try {
       AutoMLBuildSpec autoMLBuildSpec = new AutoMLBuildSpec();
       fr = parse_test_file("./smalldata/logreg/prostate_train.csv");
       autoMLBuildSpec.input_spec.training_frame = fr._key;
       autoMLBuildSpec.input_spec.response_column = "CAPSULE";
-      autoMLBuildSpec.build_control.stopping_criteria.set_max_runtime_secs(5);
+      autoMLBuildSpec.build_control.stopping_criteria.set_max_models(1);
+      autoMLBuildSpec.build_control.stopping_criteria.set_max_runtime_secs(30);
       autoMLBuildSpec.build_control.keep_cross_validation_fold_assignment = true;
 
       aml = AutoML.makeAutoML(Key.<AutoML>make(), new Date(), autoMLBuildSpec);
       AutoML.startAutoML(aml);
       aml.get();
 
-      assertTrue(aml.leader() !=null && aml.leader()._parms._keep_cross_validation_fold_assignment);
-      assertTrue(aml.leader() !=null && aml.leader()._output._cross_validation_fold_assignment_frame_id != null);
+      leader = aml.leader();
+
+      assertTrue(leader !=null && leader._parms._keep_cross_validation_fold_assignment);
+      assertNotNull(leader._output._cross_validation_fold_assignment_frame_id);
```

Comment:
This tests checks the frame key is defined but doesn't make sure it actually has any data.

Reply:
@michalkurka this is how we have been checking it so far. If id is available then we assume that our property `keep_cross_validation_fold_assignment` is working fine. See for example **pyunit_automl_args.py** test **test_keep_cross_validation_fold_assignment_enabled_with_nfolds_neq_0()**. 
Are you suggesting to change this approach of checking?

@seb I think that we should rename **test_keep_cross_validation_fold_assignment_enabled_with_nfolds_eq_0** -> **test_keep_cross_validation_fold_assignment_DISABLED_with_nfolds_eq_0**

Reply:
I would suggest adding at least

   assertNotNull(DKV.getGet(leader._output._cross_validation_fold_assignment_frame_id))

or something like that. The existence of the key doesn't imply the existence of the data.